### PR TITLE
feature: support query string as an argument in lobster codebeamer

### PR DESCRIPTION
**Pull request by mugdhadhole1**
Query string (CBQL query) can be now passed as a command line argument to the lobster codebeamer tool

### CODE REVIEW

File: lobster/tools/codebeamer/codebeamer.py [hunk 1]
```diff
@@ -152,21 +152,27 @@ def get_many_items(cb_config, item_ids):
...
-            (cb_config["base"],
-             query_id,
-             page_id,
-             cb_config["page_size"])
+        if query_id is not None:
+            url = "%s/reports/%u/items?page=%u&pageSize=%u" % \
+                (cb_config["base"],
+                    query_id,
+                    page_id,
+                    cb_config["page_size"])
```
Comment (kedarnn) [inline lobster/tools/codebeamer/codebeamer.py:171]:
```suggestion
            url = ("%s/reports/%u/items?page=%u&pageSize=%u" % 
                (cb_config["base"],
                    query_id,
                    page_id,
                    cb_config["page_size"]))
```

Comment (kedarnn) [reply lobster/tools/codebeamer/codebeamer.py:171]: Please do this wherever there is a same scenario.

File: lobster/tools/codebeamer/codebeamer.py [hunk 2]
```diff
@@ -152,23 +152,30 @@ def get_many_items(cb_config, item_ids):
...
     while total_items is None or len(rv) < total_items:
         print("Fetching page %u of query..." % page_id)
-        url = "%s/reports/%u/items?page=%u&pageSize=%u" % \
-            (cb_config["base"],
-             query_id,
-             page_id,
-             cb_config["page_size"])
-        data = query_cb_single(cb_config, url)
-        assert len(data) == 4
+        if query is not None:
```
Comment (kedarnn) [inline lobster/tools/codebeamer/codebeamer.py:166]: This check is unnecessary as on line 158 an assert statement is added to check if query is `int` or `str`.

File: packages/lobster-tool-codebeamer/README.md [hunk 3]
```diff
@@ -120,8 +120,8 @@ There are two ways you can use this tool:
 1. Download all requirements mentioned by another lobster trace (this
   way you do not get a completeness check) (using `--import-tagged`)
 
-2. Download all requirements generated by a saved codebeamer query
-  (using `--import-query`)
+2. Download all requirements generated by a saved codebeamer query ID OR a codebeamer query string (cbQL query)
```
Comment (SurajBDeore) [inline packages/lobster-tool-codebeamer/README.md:123]: please check if it is a typo error in the  cbQL query

File: lobster/tools/codebeamer/codebeamer.py [hunk 4]
```diff
@@ -525,17 +536,30 @@ def main():
...
         try:
-            query_id = int(options.import_query)
-        except ValueError:
-            ap.error("query-id must be an integer")
-        if query_id < 1:
-            ap.error("query-id must be a positive")
+            if isinstance(options.import_query, int):
+                query = int(options.import_query)
+                if query < 1:
+                    raise ValueError("query_string must be a positive integer"
```
Comment (SurajBDeore) [inline lobster/tools/codebeamer/codebeamer.py:542]: The error message is misleading; I will suggests that the issue is related to a query ID or something similar involving only integers.

**Review by kedarnn — APPROVED** [review]

**Pull request merged**